intrastat_product/models/account_move.py

```diff
@@ -0,0 +1,209 @@
+                vals = self._get_intrastat_line_vals(line)
+                if vals:
+                    line_vals.append(Command.create(vals))
+            if line_vals:
+                inv.intrastat_line_ids = line_vals
+
+    def _get_intrastat_line_vals(self, line):
+        vals = {}
+        decl_model = self.env["intrastat.product.declaration"]
+        notedict, key2label = decl_model._prepare_notedict()
+        if decl_model._is_product(line):
+            hs_code = line.product_id.get_hs_code_recursively()
+            if not hs_code:
+                return vals
+            weight, qty = decl_model._get_weight_and_supplunits(line, hs_code, notedict)
+            vals.update(
+                {
+                    "invoice_line_id": line.id,
+                    "hs_code_id": hs_code.id,
+                    "transaction_weight": weight,
+                    "transaction_suppl_unit_qty": qty,
+                    "product_origin_country_id": line.product_id.origin_country_id.id,
+                }
+            )
+        return vals
+
+    def _prepare_intrastat_line_info(self, line):
+        is_intrastat_line = bool(line._name == "account.move.intrastat.line")
+        product = line.product_id
+        return {
+            "product_id": product,
+            "hs_code_id": (
+                line.hs_code_id if is_intrastat_line else product.hs_code_id
+            ),
+            "weight": (
+                line.transaction_weight
+                if is_intrastat_line
+                else self._get_intrastat_line_vals(line)["transaction_weight"]
+            ),
+            "origin_country_id": (
+                line.product_origin_country_id
+                if is_intrastat_line
+                else product.origin_country_id
+            ),
+        }
+
+    def _get_intrastat_lines_info(self):
+        """We obtain a list of information that we will need to group at the end by
+        product and sum weight.
+        """
+        res = {}
+        for line in (
+            self.invoice_line_ids.filtered(
+                lambda x: x.product_id.hs_code_id and x.product_id.origin_country_id
+            )
+            if not self.intrastat_line_ids
+            else self.intrastat_line_ids
+        ):
+            res.setdefault(line.product_id.id, {"weight": 0})
+            vals = self._prepare_intrastat_line_info(line)
+            weight = vals.pop("weight")
+            res[line.product_id.id].update(vals)
+            res[line.product_id.id]["weight"] += weight
+        return res.values()
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    hs_code_id = fields.Many2one(
+        comodel_name="hs.code",
+        compute="_compute_hs_code_id",
+        string="Intrastat Code",
+    )
+
+    def _compute_hs_code_id(self):
+        for rec in self:
+            intrastat_line = self.move_id.intrastat_line_ids.filtered(
+                lambda r: r.invoice_line_id == rec
```

Comment:
```suggestion
                lambda r, rec=rec: r.invoice_line_id == rec
```
To fix the pre-commit error you need to like so for all cases.